pkgs/top-level/all-packages.nix

```diff
@@ -6719,12 +6719,15 @@ with pkgs;
   ipfs = callPackage ../applications/networking/ipfs {
     buildGoModule = buildGo116Module;
   };
-  ipfs-migrator = callPackage ../applications/networking/ipfs-migrator {
+  ipfs-cluster = callPackage ../applications/networking/ipfs-cluster {
     buildGoModule = buildGo116Module;
   };
-  ipfs-cluster = callPackage ../applications/networking/ipfs-cluster {
+
+  ipfs-migrator-all-fs-repo-migrations = callPackage ../applications/networking/ipfs-migrator/all-migrations.nix {
     buildGoModule = buildGo116Module;
   };
+  ipfs-migrator-unwrapped = callPackage ../applications/networking/ipfs-migrator/unwrapped.nix { };
```

Comment:
`ipfs-migrator-unwrapped` and `ipfs-migrator-all-fs-repo-migrations` are built from the same `src` but they each use a different version of `go`?

Reply:
> Go 1.17 breaks the migrations, so use Go 1.16 instead

A code comment or just pinning both would have been better ...

Reply:
Yes, `ipfs-migrator-unwrapped` works fine with the latest Go version but at least some, maybe all of the individual migrations in `ipfs-migrator-all-fs-repo-migrations` break with Go 1.17.
They all use the same `src`. Technically, it might be more correct to let fs-repo-migrations and each migration fetch its own source from a tag like https://github.com/ipfs/fs-repo-migrations/releases/tag/fs-repo-11-to-12%2Fv1.0.2 but since every tag contains the entire source code and I don't expect a the code of an individual migration to be changed without it getting a new tag, I just use the latest tag. That way we don't waste a bunch of resources fetching, unpacking and storing basically the same thing 13 times.

Reply:
I agree, I probably should have added a code comment explaining that the migrations are broken with Go 1.17.
Since fs-repo-migrations itself works fine with the new Go version, why keep it pinned at the old version as well?

Reply:
Simpler to use a consistent version when building the same `src`.

Reply:
While the `src` is the same, the `sourceRoot` is different for all 13 derivations, so in that sense, it's not the same source and the source code might as well come from different repos.

Reply:
Yes, `inherit (ipfs-migrator-unwrapped) src;` and then `sourceRoot` makes it much less clear as to why different version of `go` are used.

Just to be clear: I'm not interested in `ipfs`, etc myself, I'm one of the go maintainers in nixpkgs. Having this use two different versions of `go` make work now but it's basically just an unnecessary complication.

Reply:
> Having this use two different versions of `go` [may] work now

Do you think this might break in the future? How?

> it's basically just an unnecessary complication.

I don't see how having more overrides for using an old go version is more complicated than having fewer. But if you think using Go 1.16 for `ipfs-migrator-unwrapped` as well would be an improvement, I can make a PR adding the same override for `ipfs-migrator-unwrapped` as well. Or would adding a comment explaining that `ipfs-migrator-all-fs-repo-migrations` breaks with Go 1.17 be equally good? I don't have as much experience with the conventions for using older versions of compilers.

Reply:
> Do you think this might break in the future? How?

When we switch the default to 1.18 in a few weeks. 

> I don't see how having more overrides for using an old go version is more complicated than having fewer.

As it is currently, if it doesn't break when the default is changed to 1.18 it would using of 1.16 and 1.18 or it it does break it will be pinned to use 1.16 and 1.17. Using one version makes it easier when we are trying to update packages and remove old versions of go. 

I can do a PR for the `ipfs-migrator-unwrapped` override, I should probably add comments to the current `ipfs` overrides that were done when I did the 1.17 update.

Reply:
Sure, feel free to do the PR. Though I doubt fs-repo-migrations is actually going to break with a new Go version. It's literally just two files with less than 300 lines of code total: https://github.com/ipfs/fs-repo-migrations/tree/master/fs-repo-migrations. Not much that can go wrong there I think.
Actually `ipfs` itself seems to work fine with Go 1.17, so you can either remove the override there or pin it to Go 1.17.